utils/test_harness/tools/src/test_harness_metric.cpp

```diff
@@ -196,6 +196,31 @@ bool check_metric_type_ip(ze_device_handle_t device, std::string groupName,
   return check_metric_type_ip(groupHandle, includeExpFeature);
 }
 
+std::vector<zet_metric_group_handle_t> get_concurrent_metric_group(
+    ze_device_handle_t device,
+    std::vector<zet_metric_group_handle_t> &metricGroupHandleList) {
+
+  uint32_t concurrentGroupCount = 0;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zetDeviceGetConcurrentMetricGroupsExp(
+                device, metricGroupHandleList.size(),
+                metricGroupHandleList.data(), nullptr, &concurrentGroupCount));
```

Comment:
how do you know what size to allocate for metricGroupHandleList ?

Reply:
Its coming as a reference to get_concurrent_metric_group method right. So its already appropriately allocated.

Reply:
ok its defined by metricGroupHandleList.size()